TResNet for TIPC

caikw0602 opened the pull request:
1. Add a new "TResNetM_train_infer_python.txt" in configs for TIPC.

caikw0602 committed:
1. fix "__init__.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
1. fix "__init__.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
Merge remote-tracking branch 'origin/develop' into develop

caikw0602 committed:
1. fix "__init__.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
Merge remote-tracking branch 'origin/develop' into develop

caikw0602 committed:
1. fix "__init__.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
Merge remote-tracking branch 'origin/develop' into develop

caikw0602 committed:
1. fix "__init__.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
Merge remote-tracking branch 'origin/develop' into develop

caikw0602 committed:
1. fix "init.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
Merge remote-tracking branch 'origin/develop' into develop

caikw0602 committed:
Nothing to do

caikw0602 committed:
1. fix "init.py" of backbone to add tresnet model.
2. add tresnet model in 'tresnet.py'.
3. add "TResNetM.yaml" for training of tresnet.

caikw0602 committed:
1. Add a new "TResNetM_train_infer_python.txt" in configs for TIPC.

caikw0602 committed:
1. fix "TResNetM.yaml" for training model.

caikw0602 commented:
1. fix "TResNetM.yaml" for training model.

caikw0602 committed:
1. fix 'TResNetM_train_infer_python.txt' to run tipc successfully.

TingquanGao commented on ppcls/arch/backbone/model_zoo/tresnet.py:
```diff
@@ -0,0 +1,345 @@
+
+
+class SpaceToDepthModule(nn.Layer):
+    def __init__(self, remove_model_jit=False):
```
参数 remove_model_jit 如果没有用的话，就移除吧

TingquanGao commented on ppcls/arch/backbone/model_zoo/tresnet.py:
```diff
@@ -0,0 +1,345 @@
+        return F.conv2d(input_pad, self.filt, stride=self.stride, padding=0, groups=input.shape[1])
+
+
+class SpaceToDepth(nn.Layer):
```
SpaceToDepth 是不是可以直接整合到 SpaceToDepthModule 中呢？

caikw0602 committed:
1. remove a unused parameter "remve_model_jit"

TingquanGao commented on ppcls/arch/backbone/model_zoo/tresnet.py:
```diff
@@ -0,0 +1,346 @@
+    def forward(self, x):
+        return self.op(x)
+
+class Downsample(nn.Layer):
```
Downsample 是不是可以直接整合到 AntiAliasDownsampleLayer 中呢？

TingquanGao commented on test_tipc/configs/TResNet/TResNetM_train_infer_python.txt:
```diff
@@ -0,0 +1,54 @@
+distill_export:null
+kl_quant:null
+export2:null
+pretrained_model_url:https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/legendary_models/ResNet18_pretrained.pdparams
```
这里的路径改为这个吧：

https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/TResNetM_pretrained.pdparams

TingquanGao commented on ppcls/configs/ImageNet/TResNet/TResNetM.yaml:
```diff
@@ -0,0 +1,163 @@
+    class_id_map_file: ppcls/utils/imagenet1k_label_list.txt
+
+Metric:
+  Train:
```
这里的Train去掉吧，使用了Mixup，训练的时候无法计算Acc。具体修改可以参考：https://github.com/paddlepaddle/paddleclas/blob/release%2F2.5/ppcls/configs/ImageNet/DeiT/DeiT_tiny_patch16_224.yaml#L154-L157

TingquanGao commented:
另外，还需要：
1. 把TResNet_M_448、TResNet_L、TResNet_L_448、TResNet_XL、TResNet_XL_448这几个模型的pretrained转过来；
2. 添加以上5个模型的训练config；
3. 在TResNet组网代码中添加一下以上五个模型的接口；

caikw0602 committed:
1. Add TResNetL, TResNetXL into tresnet.py.
2. Delete TResNetM.yaml.
3. Add 6 new yamls for different vertions with different input size of TResNets for training.
4. Fix the pretrained address in TResNetM_train_infer_python.txt for tipc

caikw0602 committed:
1. fix inplanes and planes in the tresnet model

TingquanGao commented on ppcls/configs/ImageNet/TResNet/TResNetL_448.yaml:
```diff
@@ -0,0 +1,160 @@
+            to_rgb: True
+            channel_first: False
+        - RandCropImage:
+            size: 224
```
448分辨率的话，这里也要对应修改

TingquanGao commented on ppcls/configs/ImageNet/TResNet/TResNetL_224.yaml:
```diff
@@ -0,0 +1,160 @@
+      batch_transform_ops:
+        - MixupOperator:
+            alpha: 0.2
+            prob: 1.0
```
prob 字段应该不需要吧

TingquanGao commented on ppcls/configs/ImageNet/TResNet/TResNetL_224.yaml:
```diff
@@ -0,0 +1,160 @@
+  save_inference_dir: ./inference
+  # training model under @to_static
+  to_static: False
+
```
参考ImageNet/ConvNeXt/ConvNeXt_tiny.yaml添加一下EMA

caikw0602 committed:
1. fix all yaml files accordingly.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleClas?pullRequest=2636) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleClas?pullRequest=2636) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleClas?pullRequest=2636) it.</sub>
